File: backend/ClientService/src/main/java/com/example/demo/entity/RequirementTable.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "requirement_table")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RequirementTable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "req_id")
    private Integer reqId;

    @ManyToOne
    @JoinColumn(name = "client_id", nullable = false)
    private ClientTable client;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description", nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "budget", nullable = false, precision = 10, scale = 2)
    private BigDecimal budget;

    @Column(name = "status")
    private Integer status; // You can map to an enum if needed
}

File: backend/admin_panel/Models/RequirementTable.cs
using System;
using System.Collections.Generic;

namespace admin_panel.Models;

public partial class RequirementTable
{
    public int ReqId { get; set; }

    public int ClientId { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public decimal Budget { get; set; }

    public virtual ClientTable Client { get; set; } = null!;

    public virtual ICollection<ProposalsTable> ProposalsTables { get; set; } = new List<ProposalsTable>();

    public virtual ICollection<RequirementskillTable> RequirementskillTables { get; set; } = new List<RequirementskillTable>();
}
